comp/core/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -45,7 +45,14 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 		workloadmeta.NormalPriority,
 		workloadmeta.NewFilter(&filterParams),
 	)
-	resultChan := make(chan sbom.ScanResult, 2000)
+	scanner := collectors.GetContainerdScanner()
+	if scanner == nil {
+		return fmt.Errorf("error retrieving global docker scanner")
+	}
+	resultChan := scanner.Channel()
```

Comment:
`resultChan` is not use in this function except to check it is `nil`

Reply:
I pass it down here
https://github.com/DataDog/datadog-agent/blob/098c1857e3067db961d613d79a4f9e4d75dbb53f/comp/core/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go#L72

We should be able to call `scanner.Channel()` directly in the function `startScanResultHandler`. However I think that if we want to add some unit tests in the future, we'll need mocks and passing using `global` state should be avoided where we can.
WDYT ?